reframe/core/launchers/mpi.py

```diff
@@ -22,6 +22,34 @@ def command(self, job):
         return ['ibrun']
 
 
+@register_launcher('upcrun')
+class UpcrunLauncher(JobLauncher):
+    '''Launcher for UPC applications.'''
+
+    def command(self, job):
+        cmd = ['upcrun']
+        if job.num_tasks:
+            if job.num_tasks_per_node:
+                num_nodes = job.num_tasks // job.num_tasks_per_node
+                cmd += ['-N', str(num_nodes)]
```

Comment:
We need a blank line here after the if block.

Reply:
sounds good, done